sys-kernel/asahi-kernel/asahi-kernel-6.11.6_p2.ebuild

```
@@ -0,0 +1,143 @@
+# Copyright 2023-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="8"
+K_SECURITY_UNSUPPORTED="1"
+ETYPE="sources"
+K_NODRYRUN="1"
+
+RUST_MIN_VER="1.76.0"
+RUST_USEDEP='rust-src,rustfmt'
+
+inherit kernel-build rust
+
+MY_P=linux-${PV%.*}
+GENPATCHES_P="genpatches-$(ver_cut 1-2)-8"
+
+if [[ ${PV} != ${PV/_rc} ]] ; then
+	# $PV is expected to be of following form: 6.0_rc5_p1
+	MY_TAG="$(ver_cut 6)"
+	MY_BASE="$(ver_rs 2 - $(ver_cut 1-4))"
+else
+	# $PV is expected to be of following form: 5.19.0_p1
+	MY_TAG="$(ver_cut 5)"
+	if [[ "$(ver_cut 3)" == "0" ]] ; then
+		MY_BASE="$(ver_cut 1-2)"
+	else
+		MY_BASE="$(ver_cut 1-3)"
+	fi
+fi
+
+# BASE_ASAHI_TAG is the first used TAG of specific release, i.e. usually
+# the first tag of a linux 6.x or linux stable 6.x.y release
+ASAHI_TAG="asahi-${MY_BASE}-${MY_TAG}"
+
+CONFIG_VER=6.11.6-401-gentoo
+GENTOO_CONFIG_VER=g6
+FEDORA_CONFIG_DISTGIT="copr-dist-git.fedorainfracloud.org/cgit/@asahi/kernel"
+# FEDORA_CONFIG_DISTGIT="copr-dist-git.fedorainfracloud.org/cgit/ngompa/fedora-asahi-dev"
+FEDORA_CONFIG_SHA1=e5ae5b9db04b27dc69efceb35802130aec6cd375
+
+DESCRIPTION="Asahi Linux kernel sources"
+HOMEPAGE="https://asahilinux.org"
+SRC_URI="
+    https://cdn.kernel.org/pub/linux/kernel/v$(ver_cut 1).x/${MY_P}.tar.xz
+    https://github.com/AsahiLinux/linux/compare/v${MY_BASE}...${ASAHI_TAG}.patch
+        -> linux-${ASAHI_TAG}.patch
+    https://dev.gentoo.org/~mpagano/dist/genpatches/${GENPATCHES_P}.base.tar.xz
+    https://dev.gentoo.org/~mpagano/dist/genpatches/${GENPATCHES_P}.extras.tar.xz
+    https://github.com/projg2/gentoo-kernel-config/archive/${GENTOO_CONFIG_VER}.tar.gz
+        -> gentoo-kernel-config-${GENTOO_CONFIG_VER}.tar.gz
+	https://${FEDORA_CONFIG_DISTGIT}/kernel.git/plain/kernel-aarch64-16k-fedora.config?id=${FEDORA_CONFIG_SHA1}
+		-> kernel-aarch64-16k-fedora.config-${CONFIG_VER}
+"
+
+S="${WORKDIR}/${MY_P}"
+
+LICENSE="GPL-2"
+SLOT="asahi/${PVR}"
+KEYWORDS="arm64"
+
+IUSE="debug"
+
+# Rust is non-negotiable for the dist kernel
+DEPEND="
+	${DEPEND}
+	dev-util/bindgen
+	debug? ( dev-util/pahole )
+"
+
+PDEPEND="
+	~virtual/dist-kernel-${PV}
```

```suggestion
	~virtual/dist-kernel-${PV}:asahi
```

Plus similar adjustments for the other dist-kernel ebuilds to depend on `:0`.